cql3/restrictions/statement_restrictions.hh

```diff
@@ -469,6 +474,14 @@ public:
      * @return clustering key restrictions split into single column restrictions (e.g. for filtering support).
      */
     const single_column_restrictions::restrictions_map& get_single_column_clustering_key_restrictions() const;
+
+    /// Prepares internal data for evaluating index-table queries.  Must be called before
+    /// get_global_index_clustering_ranges().
+    void prepare_indexed(const schema& idx_tbl_schema, bool is_local);
+
```

Comment:
The fact that prepare-time and execution-time functions coexist in the same class is sad, but we'll tolerate it for now.